hugegraph-test/src/main/java/com/baidu/hugegraph/unit/serializer/BinarySerializerTest.java

```diff
@@ -79,4 +82,52 @@ public void testEdge() {
         Assert.assertEquals(edge2, edge);
         assertCollectionEquals(edge2.getProperties(), edge.getProperties());
     }
+
+    @Test
+    public void testVertexForPartition() {
+        BinarySerializer ser = new BinarySerializer(true, true, true);
+        HugeEdge edge = new FakeObjects().newEdge("123", "456");
+
+        BackendEntry entry1 = ser.writeVertex(edge.sourceVertex());
+        HugeVertex vertex1 = ser.readVertex(edge.graph(), entry1);
+        Assert.assertEquals(edge.sourceVertex(), vertex1);
+        assertCollectionEquals(edge.sourceVertex().getProperties(),
+                               vertex1.getProperties());
+
+        BackendEntry entry2 = ser.writeVertex(edge.targetVertex());
+        HugeVertex vertex2 = ser.readVertex(edge.graph(), entry2);
+        Assert.assertEquals(edge.targetVertex(), vertex2);
+        assertCollectionEquals(edge.targetVertex().getProperties(),
+                               vertex2.getProperties());
+
+        Whitebox.setInternalState(vertex2, "removed", true);
+        Assert.assertTrue(vertex2.removed());
+        BackendEntry entry3 = ser.writeVertex(vertex2);
+        Assert.assertEquals(0, entry3.columnsSize());
+
+        Assert.assertNull(ser.readVertex(edge.graph(), null));
+    }
+
+    @Test
+    public void testEdgeForPartition() {
+        BinarySerializer ser = new BinarySerializer(true, true, true);
+
+        FakeObjects objects = new FakeObjects();
+        HugeEdge edge1 = objects.newEdge("123", "456");
+        HugeEdge edge2 = objects.newEdge("147", "789");
+
+        BackendEntry entry1 = ser.writeEdge(edge1);
+        HugeVertex vertex1 = ser.readVertex(edge1.graph(), ser.parse(entry1));
```

Comment:
not sure why not call `ser.readVertex(edge1.graph(), entry1)`

Reply:
TODO move the logic of HBaseTable to BinarySerializer